src/main/java/com/paloaltonetworks/osc/api/PANManagerSecurityGroupInterfaceApi.java

```diff
@@ -125,26 +125,37 @@ public String findSecurityGroupInterfaceByName(String name) throws Exception {
 
         for (AddressEntry ae : this.panClient.getAddressEntries(this.devGroup)) {
 
-            if (ae == null || ae.getTagNames() == null) {
+            if (ae.getTagNames() == null) {
                 continue;
             }
 
             String sgMgrId = ae.getTagNames().stream().filter(TagToSGIdUtil::isSGTag).findFirst().orElse(null);
 
-            if (sgMgrId != null) {
-                String sgInfcMgrId = combineToSGIntefaceId(sgMgrId, this.vs.getName());
-                PANSecurityGroupInterfaceElement res = (PANSecurityGroupInterfaceElement) results.get(sgInfcMgrId);
-                if (res == null) {
-                    res = new PANSecurityGroupInterfaceElement(sgInfcMgrId, sgInfcMgrId,
-                                                               new HashSet<>(), sgInfcMgrId, sgMgrId);
-                    results.put(sgInfcMgrId, res);
-                }
-
-                Set<ManagerPolicyElement> currElements = extractPolicyElements(ae);
-                res.addManagerPolicyElements(currElements);
-            } else {
+            if (sgMgrId == null) {
                 LOG.error("Address object {} detected under {} with no security group", ae.getName(), this.devGroup);
+                continue;
+            }
+
+            // This method gets called after SG addresses have been created, but
+            // before the call to createSecurityGroupInterface (above). Without the following
+            // check, the address objects just created for the SG end up on the to-delete list
+            // inside MgrSecurityGroupInterfacesCheckMetaTask. They will be deleted right after
+            // they were created.
+            if (ae.getTagNames().size() < 2) {
+                LOG.info("Address object {} is not bound yet. Not part of any SG interface.", ae.getName());
+                continue;
             }
+
+            String sgInfcMgrId = combineToSGIntefaceId(sgMgrId, this.vs.getName());
+            PANSecurityGroupInterfaceElement res = (PANSecurityGroupInterfaceElement) results.get(sgInfcMgrId);
+            if (res == null) {
+                res = new PANSecurityGroupInterfaceElement(sgInfcMgrId, sgInfcMgrId,
+                        new HashSet<>(), sgInfcMgrId, sgMgrId);
+                results.put(sgInfcMgrId, res);
+            }
+
+            Set<ManagerPolicyElement> currElements = extractPolicyElements(ae);
+            res.addManagerPolicyElements(currElements);

```

Comment:
i think there might be a bug here. If Address A1 Contains Tag1, Tag2 and Address A2 contains only tag1

Then the security group interface will report back that it has tag1 and tag2 even though A2 only has tag1...

Reply:
That would happen if a human edited one of the addresses

Reply:
all the addresses should have the same set of tags. If there are any extra tags for the addresses we dont care, but need to make sure our policy tags are present for ALL addresses. If there are addresses missing policy tags, we need to return null so update call will be called to set the correct tags on the addresses again.